### PR TITLE
refactor mongo, pattern for statefulsets

### PR DESCRIPTION
**Pull request by jaykeshur**
…factored slightly

### CODE REVIEW

**Review by marcinc — COMMENTED** [review]
Do we need `bin/` dir in this repo?

**Review by marcinc — APPROVED** [review]
Make reference to `kd` in README. 
LGTM.

File: bin/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+info "kube api url: ${KUBE_SERVER}"
+info "namespace: ${KUBE_NAMESPACE}"
+
+export KUBE_CERTIFICATE_AUTHORITY="https://raw.githubusercontent.com/UKHomeOffice/acp-ca/master/${DRONE_DEPLOY_TO}.crt"
+
+info "deploying to environment"
+kd --check-interval=5s \
+   --timeout=5m \
+   --namespace=${KUBE_NAMESPACE} \
```
Comment (KashifSaadat) [inline bin/deploy.sh]: Nitpick: you could drop this line as kd will read the environment variable.

File: bin/env.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+export KUBE_NAMESPACE="REPLACE_ME"
+export MONGODB_STORAGE=""
+
+case "${DRONE_DEPLOY_TO}" in
+  acp-notprod)
+    export KUBE_SERVER="${KUBE_SERVER_ACP_NOTPROD}"
+    export KUBE_TOKEN="${KUBE_TOKEN_ACP_NOTPROD}"
+    ;;
+  *)
+    echo "The environment: ${DRONE_DEPLOY_TO} does is not configured"
```
Comment (KashifSaadat) [inline bin/env.sh]: Wording "does is not configured"

**Pull request merged**